kubernetes/pipeline/02configure

```diff
@@ -110,6 +110,22 @@ do
             conflicts_with $1 --docker
             MOD_OVERLAYS+=( "kubernetes/k8s-containerd.yaml" )
             ;;
+        --containerd-use-proxy) ##__OPT__type:<bool> (default=true) Set false not to use the default containerd proxy (passing a blank string to --containerd-proxy won't work)
+	    if [[ $2 != "true" ]]; then
+              MOD_PARAMS[__CONTAINERD_PROXY__]=""
+              MOD_PARAMS[__CONTAINERD_NO_PROXY__]=""
+	    fi
+	    shift
+	    ;;
+        --containerd-proxy)  #__OPT__type:<str> (default=http://squid.internal:3128)
+	    echo "XXX X: ${MOD_PARAMS[__CONTAINERD_PROXY__]} Y: $2"
+            MOD_PARAMS[__CONTAINERD_PROXY__]=$2
+            shift
+            ;;
+        --containerd-no-proxy)  #__OPT__type:<str> (default=127.0.0.1,localhost,::1,10.149.0.0/16,10.0.0.0/8,172.16.0.0/12,192.168.0.0/16)
+            MOD_PARAMS[__CONTAINERD_NO_PROXY__]=$2
+            shift
+            ;;
         --calico)
```

Comment:
We should also add options, for users to configure the docker user and password. The following config worked for me:

```
juju config containerd custom_registries='[{"url": "registry-1.docker.io", "username": "sombrafam", "password": "********"}]'
```
We should also print a warning if the user does not set those, since newer users will likely not know that this is almost mandatory and it's hard to diagnose once the deployment fails.

Reply:
I did a bunch of deployments without hitting these limits.. any idea what you need to deploy how many times to hit this?

Reply:
Think I'd prefer to address this in a separate commit to get this main one in first

Reply:
Maybe it's related to which of the pain you're connected to? I hit these many times.